pkg/interface/link/src/js/components/new.js

```diff
@@ -0,0 +1,252 @@
+
+export class NewScreen extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      idName: '',
+      groups: [],
+      ships: [],
+      idError: false,
+      inviteError: false,
+      createGroup: true
+    };
+
+    this.titleChange = this.titleChange.bind(this);
+    this.descriptionChange = this.descriptionChange.bind(this);
+    this.setInvite = this.setInvite.bind(this);
+    this.createGroupChange = this.createGroupChange.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { props, state } = this;
+
+    if (prevProps !== props) {
+      let station = `/~${window.ship}/${state.idName}`;
+      if (station in props.resources) {
+        console.log('TODO nav', '/~chat/room' + station);
+      }
+    }
+  }
+
+  titleChange(event) {
+    let asciiSafe = event.target.value.toLowerCase()
+      .replace(/[^a-z0-9-]/g, "-");
+    this.setState({
+      idName: asciiSafe + '-' + Math.floor(Math.random()*10000), // uniqueness
+      title: event.target.value
+    });
+  }
+
+  descriptionChange(event) {
+    this.setState({
+      description: event.target.value
+    });
+  }
+
+  setInvite(value) {
+    this.setState({
+      groups: value.groups,
+      ships: value.ships
+    });
+  }
+
+  createGroupChange(event) {
+    if (event.target.checked) {
+      this.setState({
+        createGroup: !!event.target.checked,
+        security: 'village'
+      });
+    } else {
+      this.setState({
+        createGroup: !!event.target.checked,
+      });
+    }
+  }
+
+  onClickCreate() {
+    const { props, state } = this;
+
+    if (!state.title) {
+      this.setState({
+        idError: true,
+        inviteError: false
+      });
+      return;
+    }
+
+    let appPath = `/${state.idName}`;
+
+    if (appPath in props.resources) {
+      this.setState({
+        inviteError: false,
+        idError: true,
+        success: false
+      });
+      return;
+    }
+
+    let isValid = true;
+    let aud = state.ships.map(mem => `~${deSig(mem.trim())}`);
+    aud.forEach((mem) => {
+      if (!urbitOb.isValidPatp(mem)) {
+        isValid = false;
+      }
+    });
+
+    if (!isValid) {
+      this.setState({
+        inviteError: true,
+        idError: false,
+        success: false
+      });
+      return;
+    }
+
+    const target = aud.length === 0
+      ? {group: state.groups[0]}
+      : {ships: aud};
+
+    if (this.textarea) {
+      this.textarea.value = '';
+    }
+
+    this.setState({
+      error: false,
+      success: true,
+      group: [],
+      ships: []
+    }, () => {
+      api.setSpinner(true);
+      //TODO account for state.createGroup
+      let submit = api.createCollection(
+        appPath,
+        state.title,
+        state.description,
+        target
+      );
+      submit.then(() => {
+        api.setSpinner(false);
+        console.log('TODO nav', `/~link/list/0${appPath}`);
```

Comment:
`props.history.push` is what you want. passing `{...props}` to new will provide that method.

Reply:
Link fe doesn't use `history` though? Not sure how to pull that in properly, but I can try mirroring what chat does, assuming that doesn't mean reworking routing entirely.

Reply:
From Link-fe:
![image](https://user-images.githubusercontent.com/20846414/75627639-f2d20800-5b9f-11ea-90eb-8cba388b44f0.png)

It comes with `react-router`. :)